### PR TITLE
fix(expect): fix the function signature of `toMatchObject()`

### PR DESCRIPTION
**Pull request by BlackGlory**
Fix https://github.com/denoland/deno_std/issues/4190

### CODE REVIEW

File: expect/_matchers.ts [hunk 1]
```diff
@@ -432,15 +432,15 @@ export function toMatch(
 
 export function toMatchObject(
   context: MatcherContext,
-  expected: Record<PropertyKey, unknown>,
+  expected: Record<PropertyKey, unknown> | Array<Record<PropertyKey, unknown>>,
```
Comment (iuioiua) [inline expect/_matchers.ts:435]:
```suggestion
  expected: Record<PropertyKey, unknown> | Record<PropertyKey, unknown>[],
```
Nit

**Review by iuioiua — CHANGES_REQUESTED** [review]
Thanks for your contribution! Can you please run `deno fmt`?

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you. @kt3k, should we also update `assertObjectMatch()` in a similar way?

**Comment by kt3k** [comment]
> LGTM. Thank you. @kt3k, should we also update assertObjectMatch() in a similar way?

I don't think so. `toMatchObject` needed to be aligned to jest, but `assertObjectMatch` doesn't need to be aligned to anything.

**Pull request merged**